Project/Source/Components/UI/ComponentEventSystem.cpp

```diff
@@ -42,34 +42,60 @@ void ComponentEventSystem::Update() {
 				SetSelected(selectable->GetID());
 			}
 		}
+		navigationTimer = Max(0.0f, navigationTimer - App->time->GetRealTimeDeltaTime());
 	}
-
-	float2 selectionDir = float2(0.f, 0.f);
 	bool keyPressed = false;
-	if (App->input->GetKey(SDL_SCANCODE_UP) == KS_DOWN) {
-		selectionDir = float2(0.f, 1.f);
-		keyPressed = true;
-	}
-	if (App->input->GetKey(SDL_SCANCODE_DOWN) == KS_DOWN) {
-		selectionDir = float2(0.f, -1.f);
-		keyPressed = true;
-	}
-	if (App->input->GetKey(SDL_SCANCODE_LEFT) == KS_DOWN) {
-		selectionDir = float2(-1.f, 0.f);
-		keyPressed = true;
-	}
-	if (App->input->GetKey(SDL_SCANCODE_RIGHT) == KS_DOWN) {
-		selectionDir = float2(1.f, 0.f);
-		keyPressed = true;
-	}
+	if (!App->userInterface->handlingSlider && navigationTimer == 0) {
+		float2 selectionDir = float2(0.f, 0.f);
+
+		if (App->input->GetKey(SDL_SCANCODE_UP) == KS_DOWN) {
+			selectionDir = float2(0.f, 1.f);
+			keyPressed = true;
+		}
+		if (App->input->GetKey(SDL_SCANCODE_DOWN) == KS_DOWN) {
+			selectionDir = float2(0.f, -1.f);
+			keyPressed = true;
+		}
+		if (App->input->GetKey(SDL_SCANCODE_LEFT) == KS_DOWN) {
+			selectionDir = float2(-1.f, 0.f);
+			keyPressed = true;
+		}
+		if (App->input->GetKey(SDL_SCANCODE_RIGHT) == KS_DOWN) {
+			selectionDir = float2(1.f, 0.f);
+			keyPressed = true;
+		}
+
+		PlayerController* controller = nullptr;
+		if (controller = App->input->GetPlayerController(0)) {
```

Comment:
You could move the assignment to the previous line to make it clearer.

Reply:
True enough